extensions/amp-analytics/0.1/requests.js

```diff
@@ -258,15 +268,26 @@ export class RequestHandler {
    */
   constructExtraUrlParamStrs_(baseUrl, extraUrlParamStrsPromise) {
     return Promise.all(extraUrlParamStrsPromise).then(paramStrs => {
-      filterSplice(paramStrs, item => {return !!item;});
-      const extraUrlParamsStr = paramStrs.join('&');
+      const extraUrlParamStrs = {
+        requestUrl: baseUrl,
+        requestBody: '',
+      };
       let requestUrl;
-      if (baseUrl.indexOf('${extraUrlParams}') >= 0) {
-        requestUrl = baseUrl.replace('${extraUrlParams}', extraUrlParamsStr);
+      if (!this.useBody_) {
```

Comment:
This will cause regression to the `GET` request. Suppose you put `useBody: true` to a `GET` request. The request will only consist the `baseUrl` here, instead of the full url with extraUrlParams. 
Because of this, I feel it is necessary to send the old `requestUrl`, the `baseUrl` and the `body` to the the `sendRequest` function. 
To do that, maybe you can create an object type called `requestBodyDef` that consists a `baseUrl` and a JsonObject, or an array of JsonObjects.  And pass both to the `sendRequest` function.  
That will also keep the code cleaner as we don't need to change the requestUrl format with different `useBody` values. 
WDYT?

Reply:
Do we want to support GET requests with body? According to HTTP spec GET body should be meaningless to the server.
My concern with this is the same as the other comment I made. To support both queryString and JSON format for the extendedParams we need to duplicate a lot of stuff or change how the extraParams expansion is done which will add a lot of complexity to the code. Specially taking into consideration that both of these cases could be interpreted as config mistakes.
 
Also, by `sendRequest`you mean `this.handler_`or `sendRequest` at amp-analytics.js level?

Reply:
I think we should let transport.js to handle the payload format, depending on different transport types. i.e., `constructExtraUrlParamStrs_` logic needs to be moved to transport.js

Reply:
but transport.js's only responsability is take a payload and sending it using the chosen method. I don't think it should be tasked with deciding how to expand its payload as well.

Reply:
It is possible that a request fallbacks to use `GET` method. 
As you can see from our code [here](https://github.com/ampproject/amphtml/blob/master/extensions/amp-analytics/0.1/transport.js#L118)
I understand that no one should use `GET` with body. But we need to make sure that when `useBody:true` is specified, the fallback case still works.
Let me know what you think.